src/main/resources/org/jenkinsci/plugins/configfiles/ConfigFilesUI/show.jelly

```diff
@@ -46,14 +46,12 @@ THE SOFTWARE.
 	<j:if test="${contentType!=null}">
 		<st:adjunct includes="
 	        org.kohsuke.stapler.codemirror.mode.${contentType.cmMode}.${contentType.cmMode},
-	        org.kohsuke.stapler.codemirror.theme.default"/>
-		<script>
-	      var editor = CodeMirror.fromTextArea(document.getElementById("config.content"), {
-	        lineNumbers: true,
-	        matchBrackets: true,
-	        mode: "${contentType.mime}",
-	        readOnly: true
-	      });
-	    </script>
+	        org.kohsuke.stapler.codemirror.theme.default,
+	        org.jenkinsci.plugins.configfiles.ConfigFilesUI.setup-codemirror"/>
+
+		<f:invisibleEntry>
+			<f:checkbox name="readOnlyFlag" checked="true"/>
+			<f:textbox field="contentType" value="${contentType.mime}"/>
+		</f:invisibleEntry>
```

Comment:
I think your approach is heavier than using e.g. a simple hidden span. You'd use those for form elements, but not to act as data holders for JavaScript.
In particular using `field` attribute of textbox will create additional data bindings, which you don't need here. You can check what the respective elements will expand to in form lib in core (https://github.com/jenkinsci/jenkins/tree/master/core/src/main/resources/lib/form).

You could stick to `invisibleEntry` if you want to as it's rather lightweight, though inside I'd for for a `span` with data attributes:
```suggestion
		<f:invisibleEntry>
		        <span data-content-type="${contentType.mime}" data-read-only="true"
		</f:invisibleEntry>
```